GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -622,10 +627,18 @@ private Result CreatePlaceholders(string directoryRelativePath, IEnumerable<Proj
                     metadata.Add("fileInfo.Name", fileInfo.Name);
                     metadata.Add("fileInfo.Size", fileInfo.Size);
                     metadata.Add("fileInfo.IsFolder", fileInfo.IsFolder);
+                    metadata.Add(nameof(result), result.ToString());
                     metadata.Add(nameof(sha), sha);
                     this.Context.Tracer.RelatedError(metadata, $"{nameof(this.CreatePlaceholders)}: Write placeholder failed");
 
-                    return result;
+                    if (result == Result.EIOError)
+                    {
+                        this.FileSystemCallbacks.OnFileConvertedToFull(childRelativePath);

```

Comment:
Minor: Could you add a comment here explaining why we're calling `OnFileConvertedToFull` and what "include mode" scenario we expect to hit this in?